### PR TITLE
redmine: Add SQLite support

### PR DESCRIPTION
**Pull request by felixsinger**
###### Description of changes

For testing and development purposes, it's not necessary to setup a SQL server. SQLite is enough for these cases.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/redmine.nix [hunk 1]
```diff
@@ -12,12 +12,14 @@ let
...
       adapter: ${cfg.database.type}
-      database: ${cfg.database.name}
+      database: ${if cfg.database.type == "sqlite3" then "${cfg.stateDir}/database.sqlite3" else cfg.database.name}
+      ${optionalString (cfg.database.type != "sqlite3") ''
       host: ${if (cfg.database.type == "postgresql" && cfg.database.socket != null) then cfg.database.socket else cfg.database.host}
       port: ${toString cfg.database.port}
       username: ${cfg.database.user}
       password: #dbpass#
       ${optionalString (cfg.database.type == "mysql2" && cfg.database.socket != null) "socket: ${cfg.database.socket}"}
+      ''}
```
Comment (felixsinger) [inline nixos/modules/services/misc/redmine.nix:22]: The indentations are wrong now. Any suggestions on how to write this differently?

```
production:
  adapter: mysql2
  database: redmine
  host: localhost
port: 3306
username: redmine
password:
socket: /run/mysqld/mysqld.sock
```

The configuration with SQLite looks good.
```
production:
  adapter: sqlite3
  database: /var/lib/redmine/database.sqlite3
```

Comment (IndeedNotJames) [reply nixos/modules/services/misc/redmine.nix:22]: YAML is a superset of JSON, meaning a YAML can contain JSON.

So you could probably just rewrite it as an attrset using either `builtins.toJSON` or `format.generate` (like `configurationYml` does) instead :)

Comment (felixsinger) [reply nixos/modules/services/misc/redmine.nix:22]: Alright, I will look into that. Thanks for the hint :)

Comment (Atemu) [reply nixos/modules/services/misc/redmine.nix:22]: We even have `lib.toYAML` which does exactly that ;)

**Comment by emilylange** [comment]
@felixsinger are you still working on this? :eyes:

**Comment by felixsinger** [comment]
I accidentally deleted the branch and thus the pull request was closed. I will continue in another pull request later.

**Comment by felixsinger** [comment]
Continuing the work in #298740

**Pull request closed without merging**